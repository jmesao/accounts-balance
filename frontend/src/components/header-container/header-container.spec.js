import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';

import HeaderContainer from './header-container.vue';

describe('HeaderContainer', () => {
  function mountHeaderContainer(props, mountOpts) {
    const localVue = createLocalVue();
    localVue.use(Vuex);

    return shallowMount(HeaderContainer, {
      localVue,
      propsData: {
        ...props,
      },
      ...mountOpts,
    });
  }

  it('should render properly', () => {
    const wrapper = mountHeaderContainer({
      title: 'Test',
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render properly with slot', () => {
    const wrapper = mountHeaderContainer({ title: 'Test' }, {
      slots: { default: '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>' },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
