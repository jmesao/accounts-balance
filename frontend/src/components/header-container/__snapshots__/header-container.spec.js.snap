// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer should render properly 1`] = `
<div
  class="header-container"
>
  <div
    class="tw-grid"
  >
    <div
      class="tw-col-full"
    >
      <h2>
        Test
      </h2>
       
    </div>
  </div>
</div>
`;

exports[`HeaderContainer should render properly with slot 1`] = `
<div
  class="header-container"
>
  <div
    class="tw-grid"
  >
    <div
      class="tw-col-full"
    >
      <h2>
        Test
      </h2>
       
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
    </div>
  </div>
</div>
`;
